pkg/cedar/core/client.go

```diff
@@ -19,11 +19,38 @@ const (
 	cedarCoreCacheDurationDefault = time.Hour * 6
 )
 
+type loggingTransport struct {
+	logger *zap.Logger
+}
+
+func (t *loggingTransport) RoundTrip(r *http.Request) (*http.Response, error) {
+	start := time.Now().UnixMilli()
+	resp, err := http.DefaultTransport.RoundTrip(r)
+	end := time.Now().UnixMilli()
+
+	t.logger.Info(
+		"Call to CEDAR core",
+		zap.String("service", "cedarcore"),
+		zap.Bool("cacheEnabled", false),
+		zap.String("method", r.Method),
+		zap.Int("status", resp.StatusCode),
+		zap.String("path", r.URL.Path),
+		zap.String("queryParams", r.URL.RawQuery),
+		zap.Int64("timeMS", end-start),
+	)
+
+	return resp, err
+}
+
 // NewClient builds the type that holds a connection to the CEDAR Core API
 func NewClient(ctx context.Context, cedarHost string, cedarAPIKey string, cedarAPIVersion string, cacheRefreshTime time.Duration, mockEnabled bool) *Client {
 	c := cache.New(cache.NoExpiration, cache.NoExpiration) // Don't expire data _or_ clean it up
 
-	hc := http.DefaultClient
+	hc := http.Client{
+		Transport: &loggingTransport{
+			logger: appcontext.ZLogger(ctx),
```

Comment:
I don't think this actually returns the logger we want to use. `NewClient` is called here in `routes.go`, and isn't the logger used _per_ request, but just the generic server logger
https://github.com/CMSgov/easi-app/blob/f276521eb04c63343a523cd7d8415076c766556d/pkg/server/routes.go#L126

Is it possible, instead, to fetch the logger off the context in the HTTP request RoundTrip method? That would probably be the correct logger, rather than attaching one to the transport.

Reply:
Honestly, poking around locally I'm not sure this is even possible as I suggested. Let's nix this feedback and just make sure we don't get the `nil` dereference, then I think we're all set!